fp-background.js

```diff
@@ -0,0 +1,450 @@
+      FolderPaneSwitcher.cachedView = null;
+      let delay = await messenger.storage.local.get("delay");
+      FolderPaneSwitcher.timer = window.setTimeout(FolderPaneSwitcher.timerCallback.notify, delay.delay, FolderPaneSwitcher);
+      //console.log("delay", delay, FolderPaneSwitcher.timer);
+    };
+
+
+    // if (FolderPaneSwitcher.timer) {
+    //   window.clearTimeout(FolderPaneSwitcher.timer);
+    //   FolderPaneSwitcher.timer = 0;
+    // };
+    // FolderPaneSwitcher.logger.debug("resettimer");
+    //    let delay = await messenger.storage.local.get("delay");
+    // FolderPaneSwitcher.timer = window.setTimeout(FolderPaneSwitcher.timerCallback.notify, delay.delay, FolderPaneSwitcher);
+    // //console.log("delay", delay, FolderPaneSwitcher.timer);
+
+  }
+
+};
+
+
+async function notifyListener(info) {
+  //   //console.log(info);
+  switch (info.command) {
+    case "onDragExit":
+      //     //console.log("bgr onDragExit");
+      FolderPaneSwitcher.onDragDrop({ type: "dragexit" });
+      break;
+    case "onDragDrop":
+      //console.log("bgr onDragDrop");
+      FolderPaneSwitcher.onDragDrop({ type: "dragdrop" });
+      break;
+    case "onDragLeave":
+      //console.log("bgr onDragLeave");
+      FolderPaneSwitcher.onDragDrop({ type: "onDragLeave" });
+      break;
+    case "onDragLeaveFolderPane":
+      FolderPaneSwitcher.onDragLeaveFolderPane({ type: "onDragLeaveFolderPane" });
+      break;
+    case "onDragEnter":
+      //console.log("bgr onDragEnter");
+      FolderPaneSwitcher.onDragEnter();
+
+      break;
+    case "onDragOver":
+      //console.log("bgr onDragOver");
+      break;
+
+    case "folderListener": //this replaces an event indicating that the message/folder drop is finished
+      //console.log("bgr folderListener", info.type);//, info.folder);
+      FolderPaneSwitcher.onDragDrop({ type: info.type });
+
+      break;
+
+  };
+};
+
+
+async function main() {
+
+
+  const windows = await messenger.windows.getAll();
+  for (let window of windows) {
+    await manipulateWindow(window);
+  }
+  messenger.windows.onCreated.addListener((window) => {
+    manipulateWindow(window);
+  });
+
+  // for future use, cuurently, event on tree is not firing
+  // messenger.messages.onMoved.addListener( async (originalMessages, movedMessages) =>
+  // {
+  //   let lastHoveredFolder = await messenger.Utilities.getLastHoveredFolder();
+  //   let newFolder = movedMessages.messages[0].folder;
+  //   //console.log("folder, ", lastHoveredFolder);
+  //   //console.log("eqhul", lastHoveredFolder.toString() == newFolder.toString() , movedMessages);
+  //   FolderPaneSwitcher.onDragDrop({ type: "msgsMoveCopyCompleted" });
+
+  // });
+
+  //console.log("bgrd");
+  let p = await messenger.Utilities.getLegacyPrefs();
+  //console.log("p", p);
+  await messenger.NotifyTools.onNotifyBackground.removeListener(notifyListener);
+  await messenger.NotifyTools.onNotifyBackground.addListener(notifyListener);
```

Comment:
I think, if I remember corr3ectly, I am doing this because no onUninstalled or onDisabled event is fired by browser.management, and I get multiple listeners otherwise.
Just asked on element for the proper way to do this. The readme proposes to remove listeners on uninstall and disable, but how if I don't receive those events?